src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -2391,31 +2393,30 @@ private Parent updateParentAliases(Parent parent, Parent foundParent, String che
 		// Note: This is not the most efficient / concise code to do this process
 		// however due to readability and the assumingly small number of aliases 
 		// parents will have this is fine 
-
-		Set<String> parentAliasStrings = new HashSet<>();
+		HashMap<String, ParentAlias> newAliases = new HashMap<String, ParentAlias>();
 		for(ParentAlias alias : parent.getParentAliases()){
 			if (! alias.isIgnored()) {
-				parentAliasStrings.add(alias.getAliasName());
+				newAliases.put(alias.getAliasName(), alias);
 			}
 		}
-		Set<String> foundParentAliasStrings = new HashSet<>(); 
+		HashMap<String, ParentAlias> existingParentAliases = new HashMap<String, ParentAlias>();
 		for(ParentAlias alias : foundParent.getParentAliases()){
 			if (! alias.isIgnored()) {
-				foundParentAliasStrings.add(alias.getAliasName());
+				existingParentAliases.put(alias.getAliasName(), alias);
 			}
 		}
-
-		Set<ParentAlias> unionParentAliases = parent.getParentAliases();
+		// Create a set to hold the union of all unique aliases
+		// Start with the existing aliases
+		Set<ParentAlias> unionParentAliases = foundParent.getParentAliases();
+		// Find any new aliases not already in the old set, and add them
+		Set<String> newAliasStrings = newAliases.keySet().stream().filter(e -> 
+			!existingParentAliases.keySet().contains(e)).collect(Collectors.toSet());
+		for(String newAliasString : newAliasStrings) {
+			unionParentAliases.add(newAliases.get(newAliasString));
+		}
 		// Parent Would Be Found Parent w/ Appropriate Updates
 		parent = foundParent;
 		// Update Parent Object to Have "Union" of All Aliases 
-		for(ParentAlias oldAlias : parent.getParentAliases()) {
```

Comment:
Then we loop through the existing aliases, and only add ones whose string is not found in the new aliases.